### PR TITLE
Cleans up pool code

### PR DESCRIPTION
**Pull request by yoyobatty**
[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. This includes, new features, sprites, sounds, balance changes, admin tools, map edits, removals, big refactors, config changes, hosting changes and important fixes. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs)

:cl: YoYoBatty
fix: Fixes pool controller wires being unable to be interacted with after a certain wire was cut.
refactor: Refactors pool controller to use screwdriver_act.
tweak: Pool controller doesn't process endless when broken or off
/:cl:

[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)

### CODE REVIEW

File: hippiestation/code/modules/pool/pool_controller.dm [hunk 1]
```diff
@@ -50,10 +50,13 @@
 			message_admins("[key_name_admin(user)] emagged the poolcontroller")
 
 /obj/machinery/poolcontroller/attackby(obj/item/W, mob/user)
+	if(issilicon(user) && get_dist(src,user)>1)
```
Comment (Jujumatic) [inline hippiestation/code/modules/pool/pool_controller.dm]: what dose this do it looks kinda funky

Comment (ghost) [reply hippiestation/code/modules/pool/pool_controller.dm]: pretty sure there's an adjacency proc that does the get_dist() stuff

**Comment by JohnGinnane** [comment]
@yoyobatty is this ready? did you test this?

**Comment by yoyobatty** [comment]
Yeet

**Pull request merged**